src/java.base/share/classes/sun/security/pkcs/PKCS9Attribute.java

```diff
@@ -367,34 +315,28 @@ public class PKCS9Attribute implements DerEncoder {
      * if the <code>value</code> has the wrong type.
      */
     public PKCS9Attribute(ObjectIdentifier oid, Object value)
-    throws IllegalArgumentException {
-        init(oid, value);
-    }
-
-    private void init(ObjectIdentifier oid, Object value)
         throws IllegalArgumentException {
 
         this.oid = oid;
-        index = indexOf(oid, PKCS9_OIDS, 1);
-        Class<?> clazz = index == -1 ? BYTE_ARRAY_CLASS: VALUE_CLASSES[index];
+        info = oidMap.get(oid);
```

Comment:
What if `info` is `null` (there is no entry for the specified OID)? The prior code set the value class to `byte[]` in that case. I think we want to preserve that behavior.

Reply:
So, it seems that no test is triggering a NPE? Consider adding a test for covering this scenario?

Reply:
This was caught by `sun/security/pkcs/pkcs9/UnknownAttribute.java` and has been fixed